tendermint/tests/serialization.rs

```diff
@@ -0,0 +1,120 @@
+//! Serialization tests
+
+use serde::{Deserialize, Serialize};
+use std::time::Duration;
+use tendermint::serializers;
+
+#[derive(Serialize, Deserialize)]
+struct IntegerTests {
+    #[serde(with = "serializers::primitives::string")]
+    unsigned: u64,
+
+    #[serde(with = "serializers::primitives::string")]
+    signed: i64,
+}
+
+#[derive(Serialize, Deserialize)]
+struct DurationTests {
+    #[serde(with = "serializers::timeduration::string")]
+    duration: Duration,
+}
+
+#[derive(Serialize, Deserialize)]
+struct BytesTests {
+    #[serde(with = "serializers::bytes::hexstring")]
+    myhexbytes: Vec<u8>,
+
+    #[serde(with = "serializers::bytes::base64string")]
+    mybase64bytes: Vec<u8>,
+
+    #[serde(with = "serializers::bytes::string")]
+    stringifiedbytes: Vec<u8>,
+}
+
```

Comment:
Do you want merge all the  `_from_` and `_into_` test pairs into a single test respectively? e.g. using the roundtrip function in this repo (https://github.com/informalsystems/tendermint-rs/pull/248#issuecomment-624605718)? This would ensure we can go from json -> struct -> json and all matches expectations.

Reply:
In the case of nullable `Vec<u8>` we actually don't get the same object back if we do a roundtrip. `null -> [] -> ""` is what happens because of how we interpret the field. But the idea is good, I'd like to play around with it.

Shall I do it in this PR or open an issue?

Reply:
Note: the nullable `Vec<u8>` is implemented in the next serialization step. So it won't show up here. I'd prefer to rewrite the tests after this PR. I'll open an issue about it.

Reply:
Sounds good!

Reply:
Opened issue #261 to track implementing test pairs.